### ISSUE TITLE
jsonimpl.py fails with Python <=2.5 if json-py is installed

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#488]
json-py seems to be yet another JSON implementation for python. See http://sourceforge.net/projects/json-py/.

If it's installed, util/jsonimpl.py fails to import simplejson as `import json` imports the json-py library.

Suggested fix in util/jsonimpl.py:

```
 try:
     import json
+    if not hasattr(json, 'JSONEncoder'):
+        raise ImportError
     JSONEncoder = json.JSONEncoder
 except ImportError:
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/488
- Originally reported by: Anonymous
- Originally created at: 2010-08-05T11:37:11.627

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-08-05 10:05:27+00:00_

Thanks, fixed in <<cset 29547f798172a34e00fded1da2f9dc0b0f31bc09>>.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-03-31 13:57:21+00:00_

Removing version: 0.6.5 (automated comment)
